### ISSUE TITLE
Stripe integration

### ISSUE THREAD

**Issue by djac** [alberto56/hatch-ahora#24]

**Comment by alberto56** [comment]
Thanks @djac 

There will be no 30 days free trial so you can forget that part, thanks!

Please update your progress here, thanks.

**Comment by djac** [comment]
Thanks @alberto56 !

I don't believe I've received Stripe access yet. Any chance you could ping Colleen for me? Thanks!

**Comment by djac** [comment]
@alberto56 attached are the necessary files for a simple Stripe Donation form.

![image](https://user-images.githubusercontent.com/459569/116329127-92852900-a798-11eb-9ec6-8da73e939249.png)
Screenshot of the donation form.

Note the files currently link to my test Stripe environment.

index.html will require the following to be updated to use Ahora's Stripe API details:
1) Stripe PUBLISHABLE_KEY
2) All pricing and product id's to match the "Donation" product that needs to be created in Stripe.

Text and images should be updated in success.html and canceled.html to match Ahora's branding.

[ahora-stripe-clientonly-checkout.zip](https://github.com/alberto56/hatch-ahora/files/6388354/ahora-stripe-clientonly-checkout.zip)

**Comment by alberto56** [comment]
Thanks @djac I opened an email thread with Colleen and you to make sure you confirm access to Ahora's stripe account.

**Comment by alberto56** [comment]
@djac re: Stripe PUBLISHABLE_KEY, if you have access to this please paste it here.

My understanding is that this is something that can be public.

**Comment by djac** [comment]
@alberto56 that is correct, PUBLISHABLE_KEY can be made public. It is the secret key that cannot be shared.

Here are the keys for test and live environments:
TEST: `pk_test_51HvqsVDY9YMnm4cO2KnD8936EzwJUDMVqpAhTl3cZCnwdxDoBcjNnPSk7uZ3MPIM37rKDsbstZPsP7M5QuQN4sfK00Z75LbSM8`
LIVE: `pk_live_51HvqsVDY9YMnm4cOxbdPgRCzuFOYjw7bxwH2PlmjztYTrsQY1bt8HmAbc3mzY1w3hs8Kr9iswqrQX4zQa2xfdChv009mNENYdu`

We'll need to setup a product (named "Donation"?) and some amounts that can be used to make a donation. Do the amounts in the screenshot I provided make sense? Let me know and I can set up the products!

**Comment by alberto56** [comment]
Can we please have these amounts in addition to "custom"?

<img width="643" alt="Screen Shot 2021-05-02 at 8 36 47 AM" src="https://user-images.githubusercontent.com/808265/116813388-9232b800-ab21-11eb-9221-fe9b0e944b77.png">

**Comment by alberto56** [comment]
@djac for the above amounts, you can paste here only the IDs for each amount, I have the page code already, thanks.

**Comment by alberto56** [comment]
Hi @djac 

* I put the code you sent me at https://github.com/alberto56/hatch-ahora/tree/9/docs
* at https://github.com/alberto56/hatch-ahora/blob/9/docs/index.html#L61 I have put the test API key
* I made https://github.com/alberto56/hatch-ahora/blob/9/docs available at https://ahora-stripe-test2.dcycleproject.org

At this point if you go to https://ahora-stripe-test2.dcycleproject.org and click on Donate $10 I get:

> No such SKU

Can you please update https://github.com/alberto56/hatch-ahora/blob/9/docs until the donation works at https://ahora-stripe-test2.dcycleproject.org, thanks!

**Comment by alberto56** [comment]
@djac as a follow-up to the above comment

<img width="721" alt="Screen Shot 2021-05-02 at 10 50 49 AM" src="https://user-images.githubusercontent.com/808265/116817438-50f7d380-ab34-11eb-9295-4d58987fc3fa.png">

**Comment by djac** [comment]
@alberto56 I've updated the page and it now works for the donation amounts provided.

Note, we cannot do a custom price using the client-only Stripe integration, which is what we are using for the donation page. For a custom amount, we'd need to use a client+server integration.

Perhaps there can be a note to contact Ahora if they'd like to donate a custom amount? A manual invoice could be created in Stripe for a given email address for custom donation amounts.

Test Stripe card information can be found here: https://stripe.com/docs/testing

**Comment by djac** [comment]
@alberto56 thanks for the test fixes!

The following items could be handled in follow-up tickets:

- [ ] Form for canceling subscriptions
- [ ] Form to update card on file (in case of failed payments)
- [ ] Hide Stripe Customer ID user field
- [ ] Handle cancelations due to failed payments or are subscription cancelations enough?
- [ ] Differentiate between inactive and active subscriptions that are canceled
- [ ] Create incomplete subscriptions in Stripe later in the checkout process. Is this desired and is it possible?
